### PR TITLE
ci: Use alpine for golangci-lint image

### PR DESCRIPTION
**Pull request by KasonBraley**
Provides a speed improvement when running the lint pipelines. About a 40 second improvement for the sdk/go lint run.

sdk/go:
Before(4m 22s):
https://github.com/KasonBraley/dagger/actions/runs/5251099871/jobs/9485711433

With this change(3m 40s):
https://github.com/KasonBraley/dagger/actions/runs/5251069261/jobs/9485647687

### CODE REVIEW

**Review by TomChv — APPROVED** [review]
Good catch, this change make sense to me, alpine image is enough for this execution.

We can merge this PR after another approval.

/cc @sipsma @vito @gerhard

**Review by vito — APPROVED** [review]

**Comment by vito** [comment]
Thanks!

**Pull request merged**
